mesh/amqp.c

```diff
@@ -609,6 +631,18 @@ static void control_message_handler(struct amqp_thread_context *context)
 							msg.publish.size);
 			return;
 
+		case SUBSCRIBE:
+			l_queue_push_tail(context->subscriptions,
+					l_strdup(msg.subscribe.pattern));
+
+			if (context->amqp_state == MESH_AMQP_STATE_CONNECTED) {
+				destroy_connection(context);
```

Comment:
This means that calling this api will immediately cause a reconnect... Im not a fan of such violence ;)